index.js

```diff
@@ -319,36 +319,37 @@ const captureWebsite = async (input, options) => {
 	}
 
 	if (screenshotOptions.fullPage) {
-		// Get the height of the rendered page
-		const bodyHandle = await page.$('body');
-		const bodyBoundingHeight = await bodyHandle.boundingBox();
-		await bodyHandle.dispose();
-
-		// Scroll one viewport at a time, pausing to let content load
-		const viewportHeight = viewportOptions.height;
-		let viewportIncrement = 0;
-		while (viewportIncrement + viewportHeight < bodyBoundingHeight) {
-			const navigationPromise = page.waitForNavigation({waitUntil: 'networkidle0'});
-			/* eslint-disable no-await-in-loop */
-			await page.evaluate(_viewportHeight => {
-				/* eslint-disable no-undef */
-				window.scrollBy(0, _viewportHeight);
-				/* eslint-enable no-undef */
-			}, viewportHeight);
-			await navigationPromise;
-			/* eslint-enable no-await-in-loop */
-			viewportIncrement += viewportHeight;
+		const autoScroll = async () => {
+			const isBottom = await page.evaluate(() => {
+				window.scrollBy(0, window.innerHeight);
+				return window.scrollY >= document.body.clientHeight - window.innerHeight;
+			});
+
+			await page.waitForFunction(imagesHaveLoaded, {timeout: timeoutInSeconds});
+
+			return !isBottom;
+		};
+
+		while (await autoScroll()) { /* eslint-disable-line no-await-in-loop */
+			// noop
 		}
 
-		// Scroll back to top
-		await page.evaluate(_ => {
-			/* eslint-disable no-undef */
-			window.scrollTo(0, 0);
-			/* eslint-enable no-undef */
+		// Workaround for chromium height limitations:  https://bugs.chromium.org/p/chromium/issues/detail?id=770769#c12
+		const height = await page.evaluate(() => document.documentElement.scrollHeight);
+		const maxTextureSize = await page.evaluate(() => {
+			const canvas = document.createElement('canvas');
+			const webGL = canvas.getContext('webgl');
+			return webGL.getParameter(webGL.MAX_TEXTURE_SIZE);
 		});
```

Comment:
`MAX_TEXTURE_SIZE` is the number of pixels we can go in any dimmension before duplicate rendering will occur.  This is determined at runtime as it can/will be different per environment.  Using this, we can adjust our `deviceScaleFactor` to fit the page within this dimension.

**Note:**  The same issue can technically happen horizontally.  If there's a use case out there, we should scroll and scale in both dimensions ( maybe a follow-up PR? ).